engine/src/main/java/org/hibernate/search/util/impl/ScopedAnalyzerReference.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.hibernate.search.analyzer.impl.AnalyzerReference;
+import org.hibernate.search.analyzer.impl.LuceneAnalyzerReference;
+
+/**
+ * A wrapper containing all the analyzers for a given class.
+ *
+ * @author Davide D'Alto
+ */
+public final class ScopedAnalyzerReference implements AnalyzerReference {
+
+	private final Map<String, AnalyzerReference> analyzers = new HashMap<>();
```

Comment:
That's exactly the same that I was referring to when mentioning thread dodginess on HipChat ;-)
It happens to be used by different threads but it seems fine for the specific use case. Definitely something to improve on, but also something that Davide just inherited from the original code here.